RDKOSS-371: Changes required in meta-rdk-oss-reference to build RDKB

Arjun-DR opened the pull request:
Reason for change: Bring in required changes to build RDKB stack

Signed-off-by: Arjun <arjun_daasuramdass@comcast.com>
(cherry picked from commit 134817eaa83f6c661673c2b4056951b5826d8244)

Update package_revisions_oss.inc

Reason for change: Set the preferred version for dnsmasq so that it won't pick the wrong version

(cherry picked from commit 386c7857df08966265357943386cb53ef94d8409)

Arjun-DR committed:
RDKOSS-371:Changes required in meta-rdk-oss-reference

Reason for change: Bring in required changes to build RDKB stack

Signed-off-by: Arjun <arjun_daasuramdass@comcast.com>
(cherry picked from commit 134817eaa83f6c661673c2b4056951b5826d8244)

Update package_revisions_oss.inc

Reason for change: Set the preferred version for dnsmasq so that it won't pick the wrong version

(cherry picked from commit 386c7857df08966265357943386cb53ef94d8409)

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 0 **violation**s, 0 files pending approval, 90 files pending identification.
* Protex Server Path: /home/blackduck/github/meta-rdk-oss-reference/261/rdkcentral/meta-rdk-oss-reference

* Commit: 3fa9d5e5f6ab32bca5bbbd1b6d035ec21e988431

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/5e1c743a29d29dce33d1335537f89fce)'

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 0 **violation**s, 1 file pending approval, 0 files pending identification.
* Protex Server Path: /home/blackduck/github/meta-rdk-oss-reference/261/rdkcentral/meta-rdk-oss-reference

* Commit: 3fa9d5e5f6ab32bca5bbbd1b6d035ec21e988431

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/2ea78a3e91ab751ab0a40f966c15f4e5)'

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 0 **violation**s, 1 file pending approval, 0 files pending identification.
* Protex Server Path: /home/blackduck/github/meta-rdk-oss-reference/261/rdkcentral/meta-rdk-oss-reference

* Commit: 3fa9d5e5f6ab32bca5bbbd1b6d035ec21e988431

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/e0eb6905a29c4c2bd9fae9b85512a3ed)'

mhughesacn commented:
Hi @Arjun-DR : I have cleared off lots of the complaints in the tools but please will you fix the following:
- Check that all patches have at least a Source: field giving the source of the changed code in the patch.
- Please remove all headers from bb and bbappend files, as the default will then apply.  Confidential headers should not be used in any case as this repo is opensourced.
- Please remove all License: fields in patch headers as we no longer use them.  (The license of the patch follows what is being patched.)
Thank you!

maniselva006c reviewed (commented):
This is huge change and difficult to review. Could you please provide below details 
1. List all modules where the version has been updated.
2. List new modules added specifically to enable broadband functionality.
3. List modules that have been modified (e.g., new patches or recipe changes). The PR should increment the version for all existing modules that have been updated.
4. Explain the validation process for any PR to the RDKB OSS module. We need a build target like packagegroup-oss-broadband that includes all OSS modules required for an RDKB device, which will help validate OSS changes against the broadband device before merging the PR.
5.  Don't we need an OSS ARCH configuration for oss module used in RDKB ? 
6. Don't we need a new machine file  for RDKB ? 
7. Share full-stack sanity test results on at least one device from each category: TV, STB, and Broadband.
8. The feature branch is out of date. Please update

Arjun-DR committed:
RDKOSS-528:Bring bluez 5.66 for broadband support

Signed-off-by: Arjun <arjun_daasuramdass@comcast.com>
(cherry picked from commit d8f1dda7cc93431cacb62ea6e77f20ddc842bd2f)

Arjun-DR committed:
Merge pull request #286 from rdkcentral/develop

Rebase

rdkcmf-jenkins commented:
b'## Copyright scan failure
 **Commit:** baa7daf1e098af7cd325cf59baa237d899863e8b
**Report detail:**  https://gist.github.com/rdkcmf-jenkins/235b3ce43fd8caea82c7b3dc850d8377'

rdkcmf-jenkins commented:
b'## Copyright scan failure
 **Commit:** 02ba8f451803e4f1168570b1bf39c9fef44ccabc
**Report detail:**  https://gist.github.com/rdkcmf-jenkins/d8da95b915142e33d43aa87b389940f0'

Arjun-DR committed:
RDKOSS-371:Address review comments for opensourcing

Signed-off-by: Arjun <arjun_daasuramdass@comcast.com>

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 0 **violation**s, 0 files pending approval, 3 files pending identification.
* Protex Server Path: /home/blackduck/github/meta-rdk-oss-reference/261/rdkcentral/meta-rdk-oss-reference

* Commit: 32642b22d31ac772b14340600d93919b355aae38

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/3f6dad780442fa2580225b5e97e7e9d0)'

Arjun-DR committed:
RDKOSS-371: Add arch and packagegroup for broadband

Reason for change: Add proper arch for respective default tune and add a
new packagegroup for broadband oss components. Also replace "_kirkstone"
,  "_broadband" with ":" to support new override syntax

Signed-off-by: Arjun <arjun_daasuramdass@comcast.com>

Arjun-DR committed:
Merge pull request #290 from rdkcentral/develop

Rebase

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 0 **violation**s, 0 files pending approval, 4 files pending identification.
* Protex Server Path: /home/blackduck/github/meta-rdk-oss-reference/261/rdkcentral/meta-rdk-oss-reference

* Commit: 3750f82eac9af97a697d92b6fbdeac15d72812a2

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/6dea7fb7e67254f7006ef812eee9c779)'

mhughesacn reviewed (commented):
Hi @Arjun-DR : Thank you for making changes here.  There are still 3 headerless patches and some problem copyrights.

maniselva006c commented on recipes-common/lighttpd/lighttpd/lighttpd_jst.conf_broadband:
```diff
@@ -0,0 +1,326 @@
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
```
how is this conf file used ?

maniselva006c commented on recipes-connectivity/bluetooth/bleadvhci.bb:
```diff
@@ -0,0 +1,23 @@
+DESCRIPTION = "Bluetooth LE Application to advertise on boot-up"
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/Apache-2.0;md5=89aea4e17d99a7cacdbeed46a0096b10"
```
which oe repo provides this recipe ?

maniselva006c commented on recipes-connectivity/socat/socat_%.bbappend:
```diff
@@ -0,0 +1,25 @@
+SRC_URI:append = " \
+        file://socat_restrict_localhost.patch \
+        file://socat_windows_config_pty.patch \
+        "
```
Are these changes generic , applicable for video devices?

maniselva006c commented on recipes-core/dropbear/dropbear_%.bbappend:
```diff
@@ -6,3 +6,9 @@ do_install:append() {
 	fi
 }
+
+DEPENDS:append:broadband = " telemetry"
```
telemetry is a MW layer component. oss layer cannot dependent on components from MW.

Arjun-DR replied:
Done.

maniselva006c commented on recipes-core/dropbear/dropbear_2020.%.bbappend:
```diff
@@ -9,3 +9,11 @@ SRC_URI:append = " file://dropbear_2019-verbose.patch \
+
+SRC_URI:append:broadband = " file://dropbear_logging_fix.patch"
+
+CFLAGS:append:broadband = " -DRDK_BROADBAND"
```
how this flag is used ?

Arjun-DR replied:
removed the flag.

maniselva006c commented on recipes-core/systemd/systemd_250.%.bbappend:
```diff
@@ -1,140 +1,199 @@
+            file://0001-Added-Extra-information-for-NTP-Status-250.patch  \
+            file://systemd250-ntp-event-trigger.patch \
+            file://0001-In-our-echo-system-we-are-managing-last-known-good-t-250.patch \
+            ${@bb.utils.contains('DISTRO_FEATURES', 'systimemgr', ' file://systemtimemgr_ntp.patch', '', d)} \
```
can we default this patch ?

maniselva006c commented on recipes-mac/AppArmor-cache/apparmor-cache_1.0.0.bb:
```diff
@@ -0,0 +1,95 @@
+FILES:${PN} += "/etc/apparmor/aa_profiles/"
+
+FILES:${PN}-native += "${base_sbindir}/apparmor_parser"
+FILES:${PN}-native += "${libdir}/features"
```
This change is currently maintained in cpc security layer with additional changes for RDKE. please check with @Ananth916  @muraliselvaraj2020  @leenaS-d  for moving it to oss layer.

Arjun-DR replied:
Yea, removed it.

maniselva006c commented on recipes-protocols/net-snmp/net-snmp-5.9.3/DELIA-40163.patch:
```diff
@@ -0,0 +1,45 @@
+Index: net-snmp-5.8/snmplib/cert_util.c
+===================================================================
+--- net-snmp-5.8.orig/snmplib/cert_util.c
++++ net-snmp-5.8/snmplib/cert_util.c
```
please move the comcast specific MIBs to cpc layer . keep only the generic changes

maniselva006c commented on recipes-support/aker/aker_git.bb:
```diff
@@ -0,0 +1,51 @@
+SUMMARY = "This receipe compiles and builds aker."
+SECTION = "libs"
+DESCRIPTION = "Module for selectively blocking MAC addresses based on a succinct schedule in MsgPack"
+HOMEPAGE = "https://github.com/Comcast/aker"
```
is it a comcast specific components ?

Arjun-DR replied:
Moved to other layer.

maniselva006c commented on recipes-support/ccronexpr/ccronexpr_1.0.bb:
```diff
@@ -0,0 +1,20 @@
+
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE.txt;md5=646c743a776a3dc373f94d63fb2f1a71"
+
```
which oe repo provides this recipe

maniselva006c commented on recipes-support/dlt-daemon/dlt-daemon_2.18.%.bbappend:
The cve patches are suppose to be generic. why it is applied with broadband override?

Arjun-DR commented on recipes-extended/iptables/iptables/0001-add-extentions-for-http-match-rule-specifications.patch:
```diff
@@ -0,0 +1,433 @@
++++ b/extensions/libxt_httpcookie.c
+@@ -0,0 +1,114 @@
++ /*
++ * Copyright (c) 2010 by Cisco Systems, Inc. All Rights Reserved.
```
I am not sure about it, so removed the patches and keep applying it from different layer for comcast.

mhughesacn commented on recipes-protocols/net-snmp-5.8/net-snmp/reproducibility-have-printcap.patch:
```diff
@@ -0,0 +1,19 @@
+Reproducible build: Don't check for /etc/printcap on the build machine when
```
Hi @Arjun-DR : All the more reason to give credit!  Without a source, this is claimed as our own code.  "Source: Backported from OE" is all you need for any relevant patches.

Arjun-DR replied:
sure, will do.

Arjun-DR committed:
RDKOSS-371: Address review comments

Signed-off-by: Arjun <arjun_daasuramdass@comcast.com>

rdkcmf-jenkins commented:
b'## Copyright scan failure
 **Commit:** 49c28e1cb402ac738866a658fdc12e5e0cf192a8
**Report detail:**  https://gist.github.com/rdkcmf-jenkins/150ed90a1ef36be0d992280a72289f73'

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 0 **violation**s, 0 files pending approval, 1 file pending identification.
* Protex Server Path: /home/blackduck/github/meta-rdk-oss-reference/261/rdkcentral/meta-rdk-oss-reference

* Commit: 49c28e1cb402ac738866a658fdc12e5e0cf192a8

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/a2a2d5259e46ea4fde87aadffd4eaa11)'

Arjun-DR committed:
Bring msgpack 2.1 which is required for Broadband

Signed-off-by: Arjun <arjun_daasuramdass@comcast.com>

Arjun-DR committed:
Add source field to patches

Signed-off-by: Arjun <arjun_daasuramdass@comcast.com>

maniselva006c commented on recipes-kernel/linux/rglinux%.bbappend:
```diff
@@ -0,0 +1,3 @@
+FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
+
```
please remove this file as it is not an oss provided recipe

Arjun-DR committed:
RDKOSS-371: Remove Middleware dependencies/components

Signed-off-by: Arjun <arjun_daasuramdass@comcast.com>

Arjun-DR committed:
RDKOSS-371:Remove non OE components

Signed-off-by: Arjun <arjun_daasuramdass@comcast.com>

Arjun-DR committed:
RDKOSS-371:Bump revision for yajl

Reason for change: As we have new patch for yajl, updating the revision.

Signed-off-by: Arjun <arjun_daasuramdass@comcast.com>

rdkcmf-jenkins commented:
b'## Copyright scan failure
 **Commit:** 67d1661d3826759c91e867db0023afe113458ee4
**Report detail:**  https://gist.github.com/rdkcmf-jenkins/33e469fc657c75d5a0792124b5af4759'

Arjun-DR committed:
Merge pull request #294 from rdkcentral/develop

RDKEMW-7910: Reduce opkg install thread count from 64 to 16 (#291)
